src/main/java/educonnect/logic/parser/EditCommandParser.java

```diff
@@ -41,55 +51,80 @@ public class EditCommandParser implements Parser<EditCommand> {
      */
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
-        ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_STUDENT_ID,
-                        PREFIX_EMAIL, PREFIX_TELEGRAM_HANDLE, PREFIX_LINK, PREFIX_TAG, PREFIX_TIMETABLE);
-
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_STUDENT_ID, PREFIX_EMAIL,
+            PREFIX_TELEGRAM_HANDLE, PREFIX_LINK, PREFIX_TAG, PREFIX_TIMETABLE);
+        // get identifier
+        String identifierArgs = " " + argMultimap.getPreamble();
+        ArgumentMultimap identifierArgMultimap = ArgumentTokenizer.tokenize(identifierArgs, EDIT_ID_PREFIX_EMAIL,
+            EDIT_ID_PREFIX_STUDENT_ID, EDIT_ID_PREFIX_INDEX, EDIT_ID_PREFIX_TELEGRAM_HANDLE);
+        // check for any valid format
+        if (argMultimap.size() < 2 && identifierArgMultimap.size() < 2) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                EditCommand.MESSAGE_USAGE));
+        }
+        // check for multiple unique identifier
+        if (identifierArgMultimap.size() != 2) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                EditCommand.MESSAGE_INVALID_IDENTIFIER));
+        }
+        // check for empty field
+        if (argMultimap.size() < 2) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_NOT_EDITED));
+        }
+        // check for duplicate field
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_STUDENT_ID, PREFIX_EMAIL, PREFIX_TELEGRAM_HANDLE,
+            PREFIX_LINK, PREFIX_TIMETABLE);
+        // check for duplicate identifier
+        identifierArgMultimap.verifyNoDuplicatePrefixesFor(EDIT_ID_PREFIX_EMAIL, EDIT_ID_PREFIX_STUDENT_ID,
+            EDIT_ID_PREFIX_INDEX, EDIT_ID_PREFIX_TELEGRAM_HANDLE);
+        // check for invalid preamble
```

Comment:
Is there a check to see if we use muitiple prefix?
eg edit s: e: .....

Reply:
it is not allowed. The code before/after that checks for multiple prefix and raise an error.